src/unistd.rs

```diff
@@ -593,6 +593,36 @@ pub fn execvp(filename: &CString, args: &[CString]) -> Result<Void> {
     Err(Error::Sys(Errno::last()))
 }
 
+/// Replace the current process image with a new one (see
+/// [fexecve(2)](http://pubs.opengroup.org/onlinepubs/9699919799/functions/fexecve.html)).
+///
+/// The `fexecve` function allows for another process to be "called" which will
+/// replace the current process image.  That is, this process becomes the new
+/// command that is run. On success, this function will not return. Instead,
+/// the new program will run until it exits.
+///
+/// This function is similar to `execve`, except that the program to be executed
+/// is referenced as a file descriptor instead of a path.
+///
+/// # Errors
+///
+/// If an error occurs, this function will return with an indication of the
+/// cause of failure.  See
+/// [fexecve(2)#errors](http://pubs.opengroup.org/onlinepubs/9699919799/functions/fexecve.html#tag_16_111_05)
+/// for a list of error conditions.
+#[cfg(not(any(target_os = "macos", target_os = "ios")))]
```

Comment:
We prefer to use white-lists here instead of black-lists as it makes it easier to support new platforms. Can you change this and the other ones below?